### DISCUSSION TITLE
Extend/override PropertyFactory or change Property buildins?

### DISCUSSION THREAD

**Discussion by tinustate** [GrapesJS/grapesjs#4878] [Q&A] [answered]
Good morning,

Is it possible to remove a specific unit for example "px" from the default properties? I would only like to support "em" (the engine rendering the final docs is a bit quirky)

I discovered PropertyFactory which defines the unitSize variable which holds the supported unit sizes for the properties, but i'm not sure how to continue.

[edit]
After a bit of fiddling and browsing the source code i came up with defining my own styleManager:
```
 styleManager: {
..
sectors: [
..
{
                        name: 'Dimensions',
                        buildProps: ['width', 'height'],
                        properties: [
                            {
                                id: "width",
                                units: ['em'],
                            },
                            {
                                id: "height",
                                units: ['em'],
                            },

                            /*'width',*/ 'height', 'max-width', 'min-height', 'margin', 'padding'],
                    },
```
Is this the way to go or is there maybe a better solution?

Kind regards,
Tinus

**Comment by artf** [comment, marked as answer]
That is one option but you can also create a plugin which [updates the built-in property](https://grapesjs.com/docs/modules/Style-manager.html#programmatic-usage)
```js
function updateBuiltInStyleProperty(editor) {
  editor.Styles.addBuiltIn('width', { units: ['em'] });
}
```
